javascript-bank/account.js

```diff
@@ -0,0 +1,37 @@
+/* exported Account */
+
+function Account(number, holder) {
+  this.number = number;
+  this.holder = holder;
+  this.transactions = [];
+}
+
+Account.prototype.deposit = function (amount) {
+  var newDeposit = new Transaction('deposit', amount);
```

Comment:
I would do this below your guard. If any of those conditions are true and this is a bad Transaction, making the Transaction object would have been a waste of work.